### ISSUE TITLE
Add a way to disable console warnings

### ISSUE THREAD

**Issue by maxpatiiuk** [Esri/calcite-design-system#10038]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

https://github.com/Esri/calcite-design-system/pull/9739 added console messages for current calcite version. These are useful in production for ensuring calcite is not loaded twice.

However, these console messages are adding noise in development. In e2e tests, it's even worse since the console message is present for each console message:

![image](https://github.com/user-attachments/assets/cffaadd2-a74c-4919-a597-e951944f6d71)

### Acceptance Criteria

1. A way to disable the console warnings when they are not necessary. For example, Lit package refers to this set on the global to decide whether warning should be emitted - https://github.com/lit/lit/blob/619449b84cb63d9c00e4316551246957c939a64b/packages/reactive-element/src/reactive-element.ts#L83-L85 - an app can define that set before importing Lit to supress the warning
2. Have Calcite ship separate production and dev mode bundles, with dev only checks in the dev bundle and production only checks in the production bundle. For example, how Lit has this configured: https://github.com/lit/lit/blob/619449b84cb63d9c00e4316551246957c939a64b/packages/reactive-element/package.json#L22-L25

### Which Component

*

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Maps SDK for JavaScript

**Comment by jcfranco** [comment]
Blocked by #10310.

**Comment by jcfranco** [comment]
Version info messages can now officially be disabled by setting `process.env.NODE_ENV = "test"` in test environments. 

**Note**: alternatively, the config version can also be overridden, but this is not recommended (see https://github.com/Esri/calcite-design-system/issues/11704).

**Comment by jcfranco** [comment]
> Have Calcite ship separate production and dev mode bundles, with dev only checks in the dev bundle and production only checks in the production bundle. For example, how Lit has this configured: [lit/lit@619449b/packages/reactive-element/package.json#L22-L25](https://github.com/lit/lit/blob/619449b84cb63d9c00e4316551246957c939a64b/packages/reactive-element/package.json#L22-L25)

I'd still like to explore this, so I'll create a separate issue for it.

**Comment by maxpatiiuk** [comment]
> I'd still like to explore this, so I'll create a separate issue for it.

Since then, Calcite moved to Lumina, and so you now have ability to include dev-only and prod-only code inside the same build, which is then eliminated from the final consumer's build.

See https://qawebgis.esri.com/components/lumina/publishing#bundling-code-conditionally, https://devtopia.esri.com/WebGIS/arcgis-web-components/issues/2035 and https://devtopia.esri.com/WebGIS/arcgis-web-components/issues/2875.

This means a separate build is no longer necessary in most cases - which simplifies the devops and improves build speed

**Comment by jcfranco** [comment]
Awesome, thanks for sharing. I'll still create an issue to leverage dev/prod modes. IIRC, we've wanted to do this for a while after we introduced dev-only log messages.
